tests/test_chatmember.py

```diff
@@ -16,112 +16,114 @@
-            'status': self.status,
-            'custom_title': custom_title,
-            'is_anonymous': True,
-            'until_date': to_timestamp(time),
-            'can_be_edited': False,
-            'can_change_info': True,
-            'can_post_messages': False,
-            'can_edit_messages': True,
-            'can_delete_messages': True,
-            'can_invite_users': False,
-            'can_restrict_members': True,
-            'can_pin_messages': False,
-            'can_promote_members': True,
-            'can_send_messages': False,
-            'can_send_media_messages': True,
-            'can_send_polls': False,
-            'can_send_other_messages': True,
-            'can_add_web_page_previews': False,
-            'can_manage_chat': True,
-            'can_manage_voice_chats': True,
-        }
-
-        chat_member = ChatMember.de_json(json_dict, bot)
-
-        assert chat_member.user == user
-        assert chat_member.status == self.status
-        assert chat_member.custom_title == custom_title
-        assert chat_member.is_anonymous is True
-        assert chat_member.can_be_edited is False
-        assert chat_member.can_change_info is True
-        assert chat_member.can_post_messages is False
-        assert chat_member.can_edit_messages is True
-        assert chat_member.can_delete_messages is True
-        assert chat_member.can_invite_users is False
-        assert chat_member.can_restrict_members is True
-        assert chat_member.can_pin_messages is False
-        assert chat_member.can_promote_members is True
-        assert chat_member.can_send_messages is False
-        assert chat_member.can_send_media_messages is True
-        assert chat_member.can_send_polls is False
-        assert chat_member.can_send_other_messages is True
-        assert chat_member.can_add_web_page_previews is False
-        assert chat_member.can_manage_chat is True
-        assert chat_member.can_manage_voice_chats is True
-
-    def test_to_dict(self, chat_member):
-        chat_member_dict = chat_member.to_dict()
-        assert isinstance(chat_member_dict, dict)
-        assert chat_member_dict['user'] == chat_member.user.to_dict()
-        assert chat_member['status'] == chat_member.status
-
-    def test_equality(self):
-        a = ChatMember(User(1, '', False), ChatMember.ADMINISTRATOR)
-        b = ChatMember(User(1, '', False), ChatMember.ADMINISTRATOR)
-        d = ChatMember(User(2, '', False), ChatMember.ADMINISTRATOR)
-        d2 = ChatMember(User(1, '', False), ChatMember.CREATOR)
+    def test_slot_behaviour(self, chatmember_types, mro_slots):
+        for attr in chatmember_types.__slots__:
+            assert getattr(chatmember_types, attr, 'err') != 'err', f"got extra slot '{attr}'"
+        assert len(mro_slots(chatmember_types)) == len(
+            set(mro_slots(chatmember_types))
+        ), "duplicate slot"
+        with pytest.raises(AttributeError):
+            chatmember_types.custom
```

Comment:
imo we should still test if setting an attribute defined by the class doesn't raise a warning, i.e, the last 2 deleted lines from this test 

``` python
chatmember_types.custom, chatmember_types.status = 'warning!', self.status
assert len(recwarn) == 1 and 'custom' in str(recwarn[0].message), recwarn.list
```

Reply:
Okay.